### ISSUE TITLE
prom-client/index has no default export

### ISSUE THREAD

**Issue by Javarome** [willsoto/nestjs-prometheus#671]
Hi, I just installed your package using npm and it complains:
```
node_modules/@willsoto/nestjs-prometheus/dist/metrics/utils.d.ts:1:8 - error TS1192: Module '"my-project/node_modules/prom-client/index"' has no default export.

1 import client from "prom-client";
```

Looking at the code: 
```ts
import client from "prom-client";
```
in your `utils` does not have a default export.
I checked the installed version of `prom-client` (which is 12.0.0) and indeed it only exports specific types.

Could you fix that, or am I missing something?

Thanks!

**Comment by willsoto** [comment]
Yup! That is working as intended. You can see an [example usage here](https://github.com/willsoto/nestjs-prometheus#usage)

**Comment by Javarome** [comment]
Sorry I don't get it. I did exactly what is specified in that example usage. prom-client installed with version 12.0.0, as well as your package. 

package.json:
```json
dependencies: {
    "@willsoto/nestjs-prometheus": "^2.0.1",
    "prom-client": "^12.0.0"
}
```
AppModule.ts:
```ts
import {PrometheusModule} from "@willsoto/nestjs-prometheus"

@Module({
  imports: [
    PrometheusModule.register({
      defaultMetrics: {
        enabled: true,
        config: {
          timeout: 5000,
          includeMethod: true,
          includePath: true,
        },
      },
    })
  ],
  providers: [],
  controllers: [],
})
export class AppModule {
}
```
This** error shows up when launching NestJS.

Could you be more specific about what I am missing? Thanks

**Comment by willsoto** [comment]
Oh apologies, I read it too quickly. The `prom-client` is not my module. The documentation for that package is [here](https://github.com/siimon/prom-client)

**Comment by willsoto** [comment]
Can you confirm that `v12` is the one actually being installed? It seems they released a v13 but I haven't had time to test any changes to see if there is an issue with that

**Comment by Javarome** [comment]
Yes I double-checked in `node-modules/prom-client/package.json` and it states `"version": "12.0.0"`
package.lock.json also states that.

**Comment by willsoto** [comment]
I'm not quite sure what the issue would be then...I'll have to take a look

**Comment by willsoto** [comment]
Do you happen to have `esModuleInterop` enabled or disabled?

**Comment by willsoto** [comment]
I think as a workaround you can disable `skipLibCheck` in your `tsconfig.json`.

**Comment by Javarome** [comment]
My esModuleInterop is disabled:
```json
{
  "compilerOptions": {
    "target": "ES2020",
    "module": "commonjs",
    "lib": [
      "ES2020"
    ],
    "outDir": "dist",
    "rootDir": "src",
    "inlineSourceMap": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "strict": true,
    "noImplicitAny": true,
    "esModuleInterop": false,
    "resolveJsonModule": true,
    "alwaysStrict": true
  },
  "include": ["./src/**/*.ts"],
  "exclude": [
    "node_modules",
    "test",
    "dist",
    "**/*spec.ts"
  ]
}
```
Did you reproduce the issue?

**Comment by willsoto** [comment]
No I didn't but I am pretty sure I know what is going on. Your tsconfig settings are incompatible with my compiled tsconfig settings - I've seen this before. Setting `skipLibCheck: false` should fix the issue on your end. I will need more time to think on this to find a proper fix.

**Comment by Javarome** [comment]
You mean `skipLibCheck: true` I guess. This does workaround the problem indeed, but I agree this is not an ideal solution.
Thanks for the tip anyway.

For the record I do not have such a problem using https://github.com/digikare/nestjs-prom
Not that I would prefer using this one, just to give you more comparison insight about the origin of the problem.

**Comment by willsoto** [comment]
Yes sorry, that is what I meant.

**Comment by willsoto** [comment]
I've released [v3.0.0](https://github.com/willsoto/nestjs-prometheus/blob/main/CHANGELOG.md#300-2020-12-23) to hopefully fix this. I am no longer using `esModuleInterop` so hopefully this means TS can figure out imports.

**Comment by Javarome** [comment]
Yes it does work with 3.0.0 and without the need for `skipLibCheck`, thanks !
